services/provider/server/server.go

```diff
@@ -275,6 +277,18 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 			"mapping":  "{}",
 		})})
 
+	monIps, err := extractMonitorIps(configmap.Data["data"])
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract monitor IPs from configmap %s: %v", monConfigMap, err)
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		ApiVersion: csiopv1a1.GroupVersion.Identifier(),
+		Kind:       "CephConnection",
```

Comment:
This becomes very verbose, and with the introduction of `ApiVersion` I can see how a copy-paste error can happen very easily.  I suggest having something like a `NewExternalResourceFor(object, client.object)` util to help in the correct reconstruction of the `ExternalResource` object.

This new function can use the schema to extract `APIVersion` and `Kind` information about the object and can extract the object name and data from the object itself.

Reply:
Will be part of different PR.